src/client/modals/source-modal/source-modal.tsx

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2017-2018 Allegro.pl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as React from "react";
+import * as CopyToClipboard from "react-copy-to-clipboard";
+import SyntaxHighlighter from "react-syntax-highlighter";
+import { githubGist } from "react-syntax-highlighter/styles/hljs";
+import { Fn } from "../../../common/utils/general/general";
+import { Button } from "../../components/button/button";
+import { Modal } from "../../components/modal/modal";
+import { STRINGS } from "../../config/constants";
+import { classNames } from "../../utils/dom/dom";
+import "./source-modal.scss";
+
+interface SourceModalProps {
```

Comment:
is it possible to reuse some modal props or class from existing sources ? Is this modal so special ?

Reply:
This modal is extracted common parts from view definition and druid query modals. So it's main purpose is to provide common modal for displaying json (later we could add other languages in props if needed). 
On the other hand, React promotes composition based on function composition instead of class extending (inheritance in normal top languages :p). For example in this case, we don't need `mandatory` property from `Modal` because this modal is never mandatory. That would get messy fast.